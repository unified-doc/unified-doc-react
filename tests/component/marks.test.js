import React from 'react';
import renderer from 'react-test-renderer';

import Doc from '../../lib/component';
import { htmlContent } from '../fixtures';

describe('marks', () => {
  it('does nothing with empty marks', () => {
    const marks = [];
    const options = {
      content: htmlContent,
      filename: 'doc.html',
      marks,
    };
    const tree = renderer.create(<Doc options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('does nothing when marks are out of range', () => {
    const marks = [{ id: 'a', start: 200, end: 400 }];
    const options = {
      content: htmlContent,
      filename: 'doc.html',
      marks,
    };
    const tree = renderer.create(<Doc options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('marks non-overlapping text nodes', () => {
    const marks = [
      { id: 'a', start: 0, end: 4 },
      { id: 'b', start: 5, end: 12 },
    ];
    const options = {
      content: htmlContent,
      filename: 'doc.html',
      marks,
    };
    const tree = renderer.create(<Doc options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('marks overlapping text nodes', () => {
    const marks = [
      { id: 'a', classNames: ['class-a'], start: 0, end: 7 },
      { id: 'b', classNames: ['class-b'], start: 5, end: 10 },
      { id: 'c', classNames: ['class-c'], start: 6, end: 15 },
    ];
    const options = {
      content: htmlContent,
      filename: 'doc.html',
      marks,
    };
    const tree = renderer.create(<Doc options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies custom class names and styles', () => {
    const marks = [
      {
        id: 'a',
        classNames: ['class-a'],
        style: { background: 'red', color: 'yellow' },
        start: 0,
        end: 4,
      },
      { id: 'b', classNames: ['class-b1', 'class-b2'], start: 5, end: 12 },
    ];
    const options = {
      content: htmlContent,
      filename: 'doc.html',
      marks,
    };
    const tree = renderer.create(<Doc options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
