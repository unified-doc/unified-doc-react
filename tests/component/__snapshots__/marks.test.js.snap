// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marks applies custom class names and styles 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        <mark
          className="class-a"
          data-mark-id="a"
          id="a"
          style={
            Object {
              "background": "red",
              "color": "yellow",
            }
          }
        >
          some
        </mark>
      </strong>
      

      <mark
        className="class-b1 class-b2"
        data-mark-id="b"
        id="b"
      >
        content
      </mark>
    </blockquote>
  </div>
</div>
`;

exports[`marks does nothing when marks are out of range 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        some
      </strong>
      
content
    </blockquote>
  </div>
</div>
`;

exports[`marks does nothing with empty marks 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        some
      </strong>
      
content
    </blockquote>
  </div>
</div>
`;

exports[`marks marks non-overlapping text nodes 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        <mark
          data-mark-id="a"
          id="a"
        >
          some
        </mark>
      </strong>
      

      <mark
        data-mark-id="b"
        id="b"
      >
        content
      </mark>
    </blockquote>
  </div>
</div>
`;

exports[`marks marks overlapping text nodes 1`] = `
<div>
  <div>
    <blockquote>
      <strong>
        <mark
          className="class-a"
          data-mark-id="a"
          id="a"
        >
          some
        </mark>
      </strong>
      <mark
        className="class-a"
        data-mark-id="a"
        id="a"
      >
        

      </mark>
      <mark
        className="class-a"
        data-mark-id="a"
      >
        <mark
          className="class-b"
          data-mark-id="b"
          id="b"
        >
          c
        </mark>
      </mark>
      <mark
        className="class-a"
        data-mark-id="a"
      >
        <mark
          className="class-b"
          data-mark-id="b"
        >
          <mark
            className="class-c"
            data-mark-id="c"
            id="c"
          >
            o
          </mark>
        </mark>
      </mark>
      <mark
        className="class-b"
        data-mark-id="b"
      >
        <mark
          className="class-c"
          data-mark-id="c"
        >
          nte
        </mark>
      </mark>
      <mark
        className="class-c"
        data-mark-id="c"
      >
        nt
      </mark>
    </blockquote>
  </div>
</div>
`;
